src/rust/engine/fs/src/store.rs

```diff
@@ -676,7 +680,7 @@ mod remote {
               )),
             })
             .map(move |maybe_bytes| {
-              maybe_bytes.map(|bytes: Vec<u8>| f(&bytes))
+              maybe_bytes.map(|bytes| f(&bytes))
```

Comment:
On a private branch improving on this PR which I haven't pushed yet, I think the only remaining copies are:
1. On fetch from CAS, we copy the n chunks into one bigger Bytes to store in lmdb. But we now do this incrementally, so we keep at most `|whole file| + |chunk|` bytes in memory, rather than 2 * |whole file|` I think this is inevitable because a) Bytes doesn't offer a merged view operator, and b) lmdb wants a byte-slice rather than an Iterator<u8> or similar.
2. On fetch from LMDB, we copy the &[u8] into a Bytes because the lifetime of the &[u8] is linked to its transaction.

Will push the update when https://github.com/pantsbuild/pants/pull/5347 gets merged :)

Reply:
I think the zero-copy is nice, but vaguely implied by using `Bytes`, and regardless probably not quite worth mentioning in the documentation :)